source/Editor.js

```diff
@@ -1680,11 +1680,11 @@ var addLinks = function ( frag, root, self ) {
 
                 child.textContent = data.slice( index, endIndex );
                 parent.insertBefore( child, node );
+                node.data = data = data.slice( endIndex );
             }
             else {
```

Comment:
The "return;" in this else statement helps to get out of the while loop. If this else is removed, it will be an endless loop when the url is undefined since the data value won't get changed and it will keep checking the same url value.